graf2d/gpadv7/v7/inc/ROOT/RAttrMap.hxx

```diff
@@ -0,0 +1,179 @@
+
+class RAttrBase;
+
+class RAttrMap {
+
+   friend class RAttrBase;
+
+public:
+
+   enum EValuesKind { kBool, kInt, kDouble, kString };
+
+   class Value_t {
+   public:
+      Value_t() = default;
+      virtual ~Value_t() = default;
+      virtual EValuesKind Kind() const = 0;
+      virtual bool Compatible(EValuesKind kind) const { return kind == Kind(); }
+      virtual bool GetBool() const { return false; }
+      virtual int GetInt() const { return 0; }
+      virtual double GetDouble() const { return 0; }
+      virtual std::string GetString() const { return ""; }
+      virtual bool IsEqual(const Value_t *) const { return false; }
+      virtual Value_t *Copy() const = 0;
+      virtual const void *GetValuePtr() const { return nullptr; }
+
+      template<typename T> T get() const;
+
+      template <typename T, typename second = void>
+      static T get_value(const Value_t *rec);
+   };
+
+   class BoolValue_t : public Value_t {
+      bool v{false}; ///< integer value
+   public:
+      explicit BoolValue_t(bool _v = false) : v(_v) {}
+      EValuesKind Kind() const final { return kBool; }
+      bool GetBool() const final { return v; }
+      Value_t *Copy() const final { return new BoolValue_t(v); }
+      bool IsEqual(const Value_t *tgt) const final { return (tgt->Kind() == kBool) && (tgt->GetBool() == v); }
+   };
+
+
+   class IntValue_t : public Value_t {
+      int v{0}; ///< integer value
+   public:
+      IntValue_t(int _v = 0) : v(_v) {}
+      EValuesKind Kind() const final { return kInt; }
+      int GetInt() const final { return v; }
+      Value_t *Copy() const final { return new IntValue_t(v); }
+      bool IsEqual(const Value_t *tgt) const final { return (tgt->Kind() == kInt) && (tgt->GetInt() == v); }
+   };
+
+   class DoubleValue_t : public Value_t {
+      double v{0}; ///< double value
+   public:
+      DoubleValue_t(double _v = 0) : v(_v) {}
+      EValuesKind Kind() const final { return kDouble; }
+      double GetDouble() const final { return v; }
+      Value_t *Copy() const final { return new DoubleValue_t(v); }
+      bool IsEqual(const Value_t *tgt) const final { return (tgt->Kind() == kDouble) && (tgt->GetDouble() == v); }
+      const void *GetValuePtr() const final { return &v; }
+   };
+
+   class StringValue_t : public Value_t {
+      std::string v; ///< string value
+   public:
+      StringValue_t(const std::string _v = "") : v(_v) {}
+      EValuesKind Kind() const final { return kString; }
+      std::string GetString() const final { return v; }
+      bool IsEqual(const Value_t *tgt) const final { return (tgt->Kind() == kString) && (tgt->GetString() == v); }
+      Value_t *Copy() const final { return new StringValue_t(v); }
+   };
+
+private:
+
+   // FIXME: due to ROOT-10306 only data member of such kind can be correctly stored by ROOT I/O
+   // Once problem fixed, one could make this container a base class
+   std::unordered_map<std::string, std::unique_ptr<Value_t>> m; ///< JSON_object
```

Comment:
Let keep it as is for the moment. This name will appear in all JSON many-many times.
I plan to modify I/O model here later anyway - this will give us freedom to use any name here.